readme.md

```diff
@@ -148,13 +149,29 @@ This package is fully typed with [TypeScript][].
 It exports an `Options` type, which specifies the interface of the accepted
 options.
 
+## Language Server features
+
+Language servers created using this package implement the following language
+server features:
+
+| Method                            | Implementation                                                                                                                                                                                                         |
+| --------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `textDocument/didChange`          | When a document is changed by the client, the language server processes it using a unified pipeline. Any messages collected are published to the client using `textDocument/publishDiagnostics`.                       |
+| `textDocument/didClose`           | When a document is closed by the client, the language server resets diagnostics by publishing an empty array using `textDocument/publishDiagnostics`.                                                                  |
+| `textDocument/didOpen`            | When a document is opened by the client, the language server processes it using a unified pipeline. Any messages collected are published to the client using `textDocument/publishDiagnostics`.                        |
+| `textDocument/formatting`         | When document formatting is requested by the client, the language server processes it using a unified pipeline. The stringified result is returned.                                                                    |
+| `workspace/didChangeWatchedFiles` | When the client signals a watched file has changed, the language server processes all open files using a unified pipeline. Any messages collected are published to the client using `textDocument/publishDiagnostics`. |
```

Comment:
Why is a table used/useful? Tables are a bit hard to maintain, and this doesn’t really seem to be “tabular data”.

I’d probably do it as a list:

```markdown
*   `textDocument/didChange`
    — when a document is changed by the client, the language server processes it
    using a unified pipeline.
    Any messages collected are published to the client using
    `textDocument/publishDiagnostics`.
*   `textDocument/didClose`
    — when a document is closed by the client, the language server resets
    diagnostics by publishing an empty array using
    `textDocument/publishDiagnostics`.
*   `textDocument/didOpen`
    — when a document is opened by the client, the language server processes it
    using a unified pipeline.
    Any messages collected are published to the client using
    `textDocument/publishDiagnostics`.
*   `textDocument/formatting`
    — when document formatting is requested by the client, the language server
    processes it using a unified pipeline.
    The stringified result is returned.
*   `workspace/didChangeWatchedFiles`
    — When the client signals a watched file has changed, the language server
    processes all open files using a unified pipeline.
    Any messages collected are published to the client using
    `textDocument/publishDiagnostics`.
```

Reply:
I have no strong feelings about this. I had to pick a format. :shrug: 

Applied your suggestion!